### DISCUSSION TITLE
Bagaimana membuat statement pada semua site url atau domain menjadi "/" saja?

### DISCUSSION THREAD

**Discussion by bungfrangki** [mecha-cms/mecha#280] [Q&A] [answered]
Misalnya kondisi site url atau domain saya adalah seperti berikut, dan di sebelahnya adalah hasil yang ingin ditampilkan:
 `<a href="http://127.0.0.1">` ==> `<a href="/">`
`<a href="http://127.0.0.1/blog">` ==> `<a href="/blog">`
`<a href="http://127.0.0.1/blog/artikel-satu">` ==> `<a href="/blog/artikel-satu">`

bisakah hal tersebut dikondisikan pada file _index.php_?
terima kasih

**Comment by taufik-nurrohman** [comment, marked as answer]
Kalau nggak salah itu sudah ditangani oleh [mecha-cms/x.minify](https://github.com/mecha-cms/x.minify). Tapi kalau tujuannya cuma untuk menghapus domain bisa sih, kira-kira begini:

~~~ php
Hook::set('content', function ($content) {
    if (!$content || false === strpos($content, '://')) {
        return $content;
    }
    $host = $_SERVER['HTTP_HOST'];
    $content = strtr($content, [
        'http://' . $host . '/' => '/',
        'http://' . $host . '?' => '?',
        'http://' . $host . '#' => '#',
        'http://' . $host . '"' => '/"',
        "http://" . $host . "'" => "/'",
    ]);
    // dst...
    // $content = strtr(...);
    return $content;
});
~~~

**Reply by bungfrangki** [reply]
oh siyap mas, nanti sy coba jalankan dengan ekstensi **x.minify** kalau sudah masuk tahap production.
untuk saat ini saya pakai function di atas dulu. terima kasih mas (y)

**Reply by bungfrangki** [reply]
saya coba modifikasi sedikit function-nya, dengan menambahkan ini, biar bisa diterapkan juga pada web yg pakai SSL:

``` php
$protocol = (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) ? "https://" : "http://";
$content = strtr($content, [
        $protocol . $host . '/' => '/',
        $protocol . $host . '?' => '?',
        $protocol . $host . '#' => '#',
        $protocol . $host . '"' => '/"',
        $protocol . $host . "'" => "/'",
    ]);
```

**Reply by taufik-nurrohman** [reply]
Sepertinya di versi lama mas, kalau di versi terakhir tidak ada fungsi untuk mengubah URL setelah saya cek.
